src/Telegram.Bot/Types/Message.cs

```diff
@@ -456,6 +490,11 @@ this switch
             { VoiceChatStarted: { } }              => MessageType.VoiceChatStarted,
             { VoiceChatEnded: { } }                => MessageType.VoiceChatEnded,
             { VoiceChatParticipantsInvited: { } }  => MessageType.VoiceChatParticipantsInvited,
+            { VideoChatScheduled: { } }            => MessageType.VideoChatScheduled,
```

Comment:
wouldn't switch process deprecated `Voice*` properties before and thus `Video*` message type will never be populated?

Reply:
No, MessageType is just for convenience and isn't present in Bot API. The actual property will be deserialized, it's just for some time both Voice* and Video* properties will be populated, but Message.Type will tell users that it has Voice* property populated. After Telegram stops send Voice* properties only Video* props will be populated and Message.Type will tell users that it's Video* type of a message

Reply:
I do not agree. Either we should put `VideoChat*` pattern before `VoiceChat*` or we should not yield `VoiceChat*` message type at all - it's misleading.